ESP32-C3: CONFIG_ESP32_PHY_MAX_TX_POWER undeclared error when building with CONFIG_BT=y

**Describe the bug**
As soon as I activate the flag CONFIG_BT I get the following error during the build of the firmware:

```
In file included from /home/user/workspace/modules/hal/espressif/components/esp_wifi/src/phy_init.c:31:
/home/user/workspace/modules/hal/espressif/zephyr/esp32c3/../../components/esp_wifi/esp32c3/include/phy_init_data.h:49:15: error: 'CONFIG_ESP32_PHY_MAX_TX_POWER' undeclared here (not in a function)
   49 |         LIMIT(CONFIG_ESP32_PHY_MAX_TX_POWER * 4, 0, 0x50),
```

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
    - ESP32-C3 DevKitM1
 - What have you tried to diagnose or workaround this issue?
    - If I enable the following flags I am able to build the application again:
      ```
      CONFIG_NETWORKING=y
      CONFIG_NET_L2_ETHERNET=y
      
      CONFIG_WIFI=y
      CONFIG_WIFI_ESP32=y
      CONFIG_MBEDTLS=y
      CONFIG_MBEDTLS_ENTROPY_ENABLED=y
      CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED=y
      CONFIG_MBEDTLS_ECP_ALL_ENABLED=y
      ```
 - ...

**To Reproduce**
Steps to reproduce the behavior:
1. Create a Zephyr application with an empty main() function
2. Run `west espressif update`
3. `west build -p auto -b esp32c3_devkitm -s my_project_folder`
4. See error

**Expected behavior**
A successful build without me having to enable the additional CONFIG flags. 

**Impact**
showstopper - Because the fimrware image becomes to large

**Logs and console output**
nothing

**Environment (please complete the following information):**
 - OS: Windows 10 with WSL2 (development environment runs inside a Docker container)
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: Zephyr (last main) commit: [806ee86](https://github.com/zephyrproject-rtos/zephyr/commit/806ee8601ca73b0de0767b2a22c533e307bcee8b)

**Additional context**
nothing

--- a/drivers/wifi/esp32/Kconfig.esp32
+++ b/drivers/wifi/esp32/Kconfig.esp32
@@ -210,18 +210,6 @@ choice ESP32_WIFI_TASK_CORE_ID
 		bool "Core 0"
 endchoice
 
-config ESP32_PHY_MAX_WIFI_TX_POWER
-	int "Max WiFi TX power (dBm)"
-	range 10 20
-	default 20
-	help
-	  Set maximum transmit power for WiFi radio. Actual transmit power for high
-	  data rates may be lower than this setting.
-
-config ESP32_PHY_MAX_TX_POWER
-	int
-	default ESP32_PHY_MAX_WIFI_TX_POWER
-
 config ESP32_WIFI_SW_COEXIST_ENABLE
 	bool
 	help

--- a/soc/riscv/esp32c3/Kconfig.soc
+++ b/soc/riscv/esp32c3/Kconfig.soc
@@ -30,4 +30,16 @@ config FLASH_BASE_ADDRESS
 	hex
 	default $(dt_node_reg_addr_hex,/soc/flash-controller@60002000/flash@0)
 
+config ESP32_PHY_MAX_WIFI_TX_POWER
+	int "Max WiFi TX power (dBm)"
+	range 10 20
+	default 20
+	help
+	  Set maximum transmit power for WiFi radio. Actual transmit power for high
+	  data rates may be lower than this setting.
+
+config ESP32_PHY_MAX_TX_POWER
+	int
+	default ESP32_PHY_MAX_WIFI_TX_POWER
+
 endif

--- a/soc/xtensa/esp32/Kconfig.soc
+++ b/soc/xtensa/esp32/Kconfig.soc
@@ -264,4 +264,16 @@ config ESP32_UNIVERSAL_MAC_ADDRESSES
 	default 2 if ESP32_UNIVERSAL_MAC_ADDRESSES_TWO
 	default 4 if ESP32_UNIVERSAL_MAC_ADDRESSES_FOUR
 
+config ESP32_PHY_MAX_WIFI_TX_POWER
+	int "Max WiFi/BLE TX power (dBm)"
+	range 10 20
+	default 20
+	help
+	  Set maximum transmit power for WiFi radio. Actual transmit power for high
+	  data rates may be lower than this setting.
+
+config ESP32_PHY_MAX_TX_POWER
+	int
+	default ESP32_PHY_MAX_WIFI_TX_POWER
+
 endif # SOC_ESP32

--- a/soc/xtensa/esp32s2/Kconfig.soc
+++ b/soc/xtensa/esp32s2/Kconfig.soc
@@ -224,4 +224,16 @@ config ESP32S2_UNIVERSAL_MAC_ADDRESSES
 	default 1 if ESP32S2_UNIVERSAL_MAC_ADDRESSES_ONE
 	default 2 if ESP32S2_UNIVERSAL_MAC_ADDRESSES_TWO
 
+config ESP32_PHY_MAX_WIFI_TX_POWER
+	int "Max WiFi TX power (dBm)"
+	range 10 20
+	default 20
+	help
+	  Set maximum transmit power for WiFi radio. Actual transmit power for high
+	  data rates may be lower than this setting.
+
+config ESP32_PHY_MAX_TX_POWER
+	int
+	default ESP32_PHY_MAX_WIFI_TX_POWER
+
 endif # SOC_ESP32S2

--- a/west.yml
+++ b/west.yml
@@ -65,7 +65,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: 5a334d5584d030d5773de2f1d489288a8625529c
+      revision: 12d88df9caf69c8b4f3ad44efc7780b0f0e50db2
       path: modules/hal/espressif
       west-commands: west/west-commands.yml
       groups:

